src/transformation/utils/annotations/collect.ts

```diff
@@ -0,0 +1,138 @@
+import * as ts from "typescript";
+import { flatMap } from "../../../utils";
+import { TransformationContext } from "../../context";
+
+export type AnnotationName = string & { _annotationNameBrand: any };
+export enum AnnotationKind {
+    Extension = "Extension",
+    MetaExtension = "MetaExtension",
+    CustomConstructor = "CustomConstructor",
+    CompileMembersOnly = "CompileMembersOnly",
+    NoResolution = "NoResolution",
+    PureAbstract = "PureAbstract",
+    Phantom = "Phantom",
+    TupleReturn = "TupleReturn",
+    LuaIterator = "LuaIterator",
+    LuaTable = "LuaTable",
+    NoSelf = "NoSelf",
+    NoSelfInFile = "NoSelfInFile",
+    Vararg = "Vararg",
+    ForRange = "ForRange",
+}
+
+function getAnnotationKindByName(name: AnnotationName): AnnotationKind;
+function getAnnotationKindByName(name: string): AnnotationKind | undefined;
```

Comment:
This line seems a duplicate to the one below, can be removed?

Reply:
It's required, last overload is used for implementation, but doesn't contribute to the type